src/test/java/org/javarosa/xform/util/test/CalendarTests.java

```diff
@@ -35,6 +55,33 @@ public void testTimesFallOnSameDate() {
         assertSameDate(endOfDay, beginningOfDay);
     }
 
+    // millis <=> date is different in every timezone
+    @Test
+    public void testConvertToNepaliString() {
```

Comment:
can you confirm these tests are failing without the changes made in `CalendarUtils` (I tried on my end but it seems to pass) ? I think we want to confirm that this test indeed reproduces the error we are trying to fix.

Reply:
@shubham1g5 the tests were considering the raw offset, which doesn't account for DST. I have updated them [334a900](https://github.com/dimagi/commcare-core/pull/1492/commits/334a9006d6b1cbb691b005c2d9ee38c1c5dd759f). Can you test this again?